src/main/java/net/dries007/tfc/world/classic/worldgen/vein/VeinRegistry.java

```diff
@@ -139,8 +143,21 @@ public void postInit()
                         VeinType vein = GSON.fromJson(entry.getValue(), VeinType.class);
                         vein.setRegistryName(properVeinName);
 
+                        // Last check before registering, make sure to not include any bogus veins
+                        if (vein.getOre() == null && vein.getOreState(Rock.GRANITE, Ore.Grade.NORMAL).getMaterial() == Material.AIR)
+                        {
+                            TerraFirmaCraft.getLog().error("Error: " + vein.getRegistryName() + " vein registered ore block is Air! Skipping.");
+                            continue;
+                        }
+
```

Comment:
There's no need for this. If the block failed to resolve to a name, it should be already caught and the vein not registered. This is only preventing the case the user wrote `"minecraft:air"` as the block, in which case why would we deny that?

Reply:
Not true. I found in my tests that if you specify something invalid (ie: `tfc:native_copper2`) the vein is parsed anyway containing air blocks. I can fix that in `VeinTypeJson` instead, but that will still prevent air blocks being registered: 
https://github.com/DisasterMoo/TerraFirmaCraft/blob/ef7fb80e6e89f5c618d761e637c5712b78170d73/src/main/java/net/dries007/tfc/util/json/VeinTypeJson.java#L111-L115

Reply:
Well that's a nuisance. In which case, [here](https://github.com/DisasterMoo/TerraFirmaCraft/blob/ef7fb80e6e89f5c618d761e637c5712b78170d73/src/main/java/net/dries007/tfc/util/json/VeinTypeJson.java#L136), switch the ordering of that call chain to:
```java
if (ForgeRegistries.BLOCKS.containsKey(oreName)) {
    Block block = ForgeRegistries.BLOCKS.getValue(oreName); // Blocks have a nonnull default key, we can ignore null checks on this
    ...
```
as that should more accurately handle this case:
 - no arbitrary restrictions on the usage
 - accurate error reporting if something invalid was specified